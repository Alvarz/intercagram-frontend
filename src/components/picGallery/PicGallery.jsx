import React from 'react'
import './css/PicGallery.scss'
import PicGalleryCtrl from './js/PicGalleryCtrl'
import Pic from '../pic/Pic'

export default class PicGallery extends PicGalleryCtrl {
  constructor (props) {
    super(props)
    this.pic = this.props.pic
    this.class = this.props.class || 'col'
    this.withDesction = this.props.withDescription
    console.log('this.withDesction', this.withDesction)
  }

  render () {
    return (
      <li key={this.pic.id} id={this.pic.id} className={this.class}>
        <Pic pic={this.pic} withDescription={this.withDescription} />
      </li>
    )
  }
}
