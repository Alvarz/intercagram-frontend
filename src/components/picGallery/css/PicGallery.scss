.desc-pic{
  margin-top:10px;
}
.pic-username{
  display:block;
  padding-top:15px;
}

.description{
  margin-top:15px;
}
