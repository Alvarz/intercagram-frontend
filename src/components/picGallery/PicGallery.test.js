import React from 'react'
import PicGallery from './PicGallery'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'

let initialState = {

}

let pic = {
  _id: 'k12h3jk12hb'
}

it('renders without crashing', () => {
  let mockStore = configureStore()

  // Store
  let store = mockStore(initialState)

  // Wrappeamos el componente
  let wrapper = shallow(<PicGallery pic={pic} store={store} />, { initialState })

  // Validamos que renderice.
  expect(wrapper).toMatchSnapshot()
})
