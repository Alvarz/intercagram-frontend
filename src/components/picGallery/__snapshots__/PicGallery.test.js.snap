// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PicGallery
    pic={
      Object {
        "_id": "k12h3jk12hb",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "k12h3jk12hb",
    "nodeType": "host",
    "props": Object {
      "children": <withRouter(Connect(Pic))
        pic={
          Object {
            "_id": "k12h3jk12hb",
          }
        }
      />,
      "className": "col",
      "id": "k12h3jk12hb",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "pic": Object {
          "_id": "k12h3jk12hb",
        },
        "withDescription": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "k12h3jk12hb",
      "nodeType": "host",
      "props": Object {
        "children": <withRouter(Connect(Pic))
          pic={
            Object {
              "_id": "k12h3jk12hb",
            }
          }
        />,
        "className": "col",
        "id": "k12h3jk12hb",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pic": Object {
            "_id": "k12h3jk12hb",
          },
          "withDescription": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "initialState": Object {},
  },
}
`;
